File: Algorithms_C++/1_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashtable;
        for (int i = 0; i < nums.size(); ++i){
            int x = nums[i];
            int y = target - x;
            if (hashtable.count(y)){
                return {hashtable[y], i};
            }
           hashtable[x] = i; 
        }
        return{};
    }
};

File: Algorithms_C#/1_Two_Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var dict = new Dictionary<int, int>();
        int n = nums.Length;
        for (int i = 0; i < n; i++){
            int y = target - nums[i];
            if (dict.ContainsKey(y) && dict[y] != i){
                return new int[]{i, dict[y]};
            }
            if(!dict.ContainsKey(nums[i])){
                dict.Add(nums[i],i);
            }
        }
        return new int[]{0,0};
    }
}